PWGJE/EMCALJetTasks/Tracks/AliAnalysisTaskEmcalJetEnergySpectrum.cxx

```diff
@@ -389,6 +397,45 @@ bool AliAnalysisTaskEmcalJetEnergySpectrum::Run(){
         fHistos->FillTH2("hQAClusterFracLeadingVsE", ptvec.E(), maxamplitude/cluster->E());
         fHistos->FillTH2("hQAClusterFracLeadingVsNcell", cluster->GetNCells(), maxamplitude/cluster->E());
       }
+
+      // 1D Cluster Histogram
+      Double_t energy(-1);
+      if(fMakeClusterHistos1D){
+        for(auto clust : clusters->all()) {
+          if(!clust->IsEMCAL()) continue;
+          if(clust->GetIsExotic()) continue;
+          if(!fIsMC) {
+            if(clust->GetTOF() < fMinTimeClusterBias || clust->GetTOF() > fMaxTimeClusterBias) continue;
+          }
+
+          // Distinguish energy definition
+          switch(fEnergyDefinition){
+          case kDefaultEnergy:
+    	      AliDebugStream(2) << GetName() << ": Using cluster energy definition: default" << std::endl;
+    	      energy = clust->E();
+    	      break;
+          case kNonLinCorrEnergy:
+    	      AliDebugStream(2) << GetName() << ": Using cluster energy definition: corrected for non-linearity" << std::endl;
+    	      energy = clust->GetNonLinCorrEnergy();
+    	      break;
+          case kHadCorrEnergy:
+    	      AliDebugStream(2) << GetName() << ": Using cluster energy definition: corrected for hadronic contribution" << std::endl;
+    	      energy = clust->GetHadCorrEnergy();
+    	      break;
+          default:
+            energy = -1;
+            break;
+          };
```

Comment:
The histogram is actually quite small, one could consider adding both non-lin and had. corr energy